### ISSUE TITLE
Crash with AAI Vehicles

### ISSUE THREAD

**Issue by jtuc** [321freddy/even-distribution#14]
While distributing ammo to [AAI Chaingunners](https://mods.factorio.com/mod/aai-vehicles-chaingunner), I got the following stacktrace when I released right-click:

```
1916.637 Error MainLoop.cpp:1183: Exception at tick 694672: The mod Even Distribution caused a non-recoverable error.
Please report this error to the mod author.

Error while running event even-distribution::on_tick (ID 0)
There is no valid object with id 1919.
stack traceback:
	__even-distribution__/scripts/visuals.lua:105: in function 'func'
	__even-distribution__/scripts/helpers.lua:141: in function 'where'
	__even-distribution__/scripts/visuals.lua:101: in function 'unmark'
	__even-distribution__/scripts/visuals.lua:115: in function 'func'
	__even-distribution__/scripts/helpers.lua:124: in function 'each'
	__even-distribution__/scripts/visuals.lua:114: in function 'unmark'
	__even-distribution__/scripts/drag.lua:173: in function 'resetCache'
	__even-distribution__/scripts/drag.lua:62: in function 'distributeItems'
	__even-distribution__/scripts/drag.lua:15: in function 'func'
	__even-distribution__/framework.lua:99: in function <__even-distribution__/framework.lua:98>
stack traceback:
	[C]: in function 'get_target'
	__even-distribution__/scripts/visuals.lua:105: in function 'func'
	__even-distribution__/scripts/helpers.lua:141: in function 'where'
	__even-distribution__/scripts/visuals.lua:101: in function 'unmark'
	__even-distribution__/scripts/visuals.lua:115: in function 'func'
	__even-distribution__/scripts/helpers.lua:124: in function 'each'
	__even-distribution__/scripts/visuals.lua:114: in function 'unmark'
	__even-distribution__/scripts/drag.lua:173: in function 'resetCache'
	__even-distribution__/scripts/drag.lua:62: in function 'distributeItems'
	__even-distribution__/scripts/drag.lua:15: in function 'func'
	__even-distribution__/framework.lua:99: in function <__even-distribution__/framework.lua:98
```

I was also using [AAI Programmable Vehicles](https://mods.factorio.com/mod/aai-programmable-vehicles), which seems to do some funky things when vehicles start/stop moving - possibly making them change IDs?
I think I got unlucky with the exact tick - I couldn't reproduce the crash easily.

**Comment by jtuc** [comment]
Actually, I can reproduce it pretty reliably in sandbox scenario; same stacktrace apart from tick/ID number.
It does seem to be when distributing items to a vehicle at the end of its movement.

Mod versions:
aai-programmable-vehicles_0.5.24
aai-signals_0.4.2
aai-vehicles-chaingunner_0.4.1
even-distribution_0.3.8

Base: 0.17.45

**Comment by 321freddy** [comment]
Should be fixed in version 0.3.9. Thanks for the report :)
